cmd/flags_core.go

```diff
@@ -51,11 +50,12 @@ func ParseCoreFlags(cmd *cobra.Command, env *Env) error {
 
 	grpcEndpoint := cmd.Flag(coreGRPCFlag).Value.String()
 	if grpcEndpoint != "" {
-		// trim ip:port from the endpoint
-		grpcEndpoint = strings.Trim(grpcEndpoint, "http://")  //nolint:staticcheck
-		grpcEndpoint = strings.Trim(grpcEndpoint, "https://") //nolint:staticcheck
-		grpcEndpoint = strings.Trim(grpcEndpoint, "tcp://")   //nolint:staticcheck
-		env.AddOptions(node.WithGRPCEndpoint(grpcEndpoint))
+		// parse ip:port from the endpoint
+		grpcURL, err := url.Parse(grpcEndpoint)
+		if err != nil {
+			return err
+		}
+		env.AddOptions(node.WithGRPCEndpoint(grpcURL.Host))
```

Comment:
maybe dumb question but what's the point of passing in the protocol (tcp/http(s)) if it is not used anyways?

Reply:
I guess I do not really understand why we are telling users that they must add the protocol only to remove it internally.
I'm wondering if users were only doing it in the first place bc of the docs?

Reply:
Users are already adding http:// and it's resulting in a lot of them not knowing why they can't connect to the RPC.

Reply:
I can make the docs say "don't add http://" but feel users will run into this issue anyways and forget to remove itand from a UX, it shouldn't be a blocker for developers looking to connect to an RPC imo.

Reply:
OK, resolving.